src/main/scala/edu/berkeley/ce/sparkrocks/RockSlicer.scala

```diff
@@ -36,26 +36,28 @@ object RockSlicer {
     val generatedInput = JointGenerator(globalOrigin, boundingBox, rockVolumeInputs, jointSetInputs)
     val starterBlocks = Seq(Block(globalOrigin, generatedInput.rockVolume))
 
+    val startTime = System.nanoTime()
     // Generate a list of initial blocks before RDD-ifying it
-    val (seedJoints, remainingJoints) = if (arguments.numProcessors > 1) {
+    val (seedBlocks, remainingJoints) = if (arguments.numPartitions > 1) {
       // Check if at least one of the input joint sets is persistent
       if (jointSetInputs.forall(_.persistence != 100.0)) {
         println("ERROR: Input joint sets must contain at least one persistent joint set to run in parallel. Rerun" +
           "analysis in serial if all joint sets are non-persistent.")
         System.exit(-1)
       }
 
-      SeedJointSelector.searchJointSets(generatedInput.jointSets, starterBlocks.head, arguments.numProcessors)
+      SeedJointSelector.generateSeedBlocks(generatedInput.jointSets, starterBlocks.head, arguments.numPartitions)
     } else {
-      (Seq.empty[Joint], generatedInput.jointSets.flatten)
+      (starterBlocks, generatedInput.jointSets.flatten)
     }
 
-    val seedBlocks = if (seedJoints.isEmpty) {
-      starterBlocks
-    } else {
-      seedJoints.foldLeft(starterBlocks) { (currentBlocks, joint) =>
-        currentBlocks.flatMap(_.cut(joint))
-      }
+    // If running without forcePartition flag, check if specified number of partitions was found
+    if (!arguments.forcePartition && seedBlocks.length != arguments.numPartitions) {
```

Comment:
I'm fine either way. Finding the exact number of partitions for a real problem is very unlikely. My thinking was that this way users would be forced to be aware of what is going on with partitions. We could add wording to the users manual that states something about using the `f` flag if they just want to proceed regardless of how many partitions are found.

What do you think would be best?